cloudos_cli/clos.py

```diff
@@ -368,52 +373,162 @@ def get_job_list(self, workspace_id, last_n_jobs=30, page=1, archived=False,
+            Filter jobs by workflow name (will be resolved to workflow ID).
+        filter_job_id : string, optional
+            Filter jobs by specific job ID.
+        filter_only_mine : bool, optional
+            Filter to show only jobs belonging to the current user.
+        filter_queue : string, optional
+            Filter jobs by queue name (local filtering, will be resolved to queue ID).
 
         Returns
         -------
         r : list
             A list of dicts, each corresponding to a jobs from the user and the workspace.
         """
+        if not workspace_id or not isinstance(workspace_id, str):
+            raise ValueError("Invalid workspace_id: must be a non-empty string")
+    
+        if last_n_jobs != 'all' and (not isinstance(last_n_jobs, int) or last_n_jobs <= 0):
+            raise ValueError("last_n_jobs must be a positive integer or 'all'")
+        
+        # Validate filter_status values
+        if filter_status:
+            valid_statuses = ['completed', 'running', 'failed', 'aborted', 'queued', 'pending', 'initializing']
+            if filter_status.lower() not in valid_statuses:
+                raise ValueError(f"Invalid filter_status '{filter_status}'. Valid values: {', '.join(valid_statuses)}")
+
         headers = {
             "Content-type": "application/json",
             "apikey": self.apikey
         }
-        if archived:
-            archived_status = "true"
-        else:
-            archived_status = "false"
-        r = retry_requests_get("{}/api/v2/jobs?teamId={}&page={}&archived.status={}".format(
-                               self.cloudos_url, workspace_id, page, archived_status),
-                               headers=headers, verify=verify)
-        if r.status_code >= 400:
-            raise BadRequestException(r)
-        content = json.loads(r.content)
-        n_jobs = len(content['jobs'])
-        if last_n_jobs == 'all':
-            jobs_to_get = n_jobs
-        elif last_n_jobs > 0:
-            jobs_to_get = last_n_jobs - n_jobs
-        else:
-            raise TypeError("[ERROR] Please select an int > 0 or 'all' for 'last_n_jobs'")
-        if jobs_to_get == 0 or n_jobs == 0:
-            return content['jobs']
-        if jobs_to_get > 0:
-            if last_n_jobs == 'all':
-                next_to_get = 'all'
-            else:
-                next_to_get = jobs_to_get
-            return content['jobs'] + self.get_job_list(workspace_id, last_n_jobs=next_to_get,
-                                                       page=page+1, archived=archived,
-                                                       verify=verify)
-        if jobs_to_get < 0:
-            return content['jobs'][:jobs_to_get]
+        
+        # Build query parameters for server-side filtering
+        params = {
+            "teamId": workspace_id,
+            "archived.status": str(archived).lower(),
+            "limit": 50,  # Use a reasonable page size
```

Comment:
```suggestion
            "limit": 100,  # Use a reasonable page size
```
Projects, workflows, and almost all new API items will have 100 pagination limit, so I think this will be consistent